Update 1.7 release notes

johrstrom opened the pull request:
https://osc.github.io/ood-documentation-test/update-1.7-release-notes

Updates for the 1.7 release notes. The only thing missing out of the list are docs on the custom classroom deployment. 

I'm sure there's more to be added to the 1.7 RNs which I can dig through/think of today.

johrstrom commented:
It should be noted this also includes updates in the develop branch because the RNs link back to them. My commits for RN update start [here](https://github.com/OSC/ood-documentation/pull/259/commits/8ecc3515889516a6a99204018e2aac082d802ea9) at commit ecc351.

treydock commented:
Should we mention Grafana support? This is section used for Owens and Pitzer in their cluster.yaml:

```yaml
    grafana:
      host: "https://grafana.osc.edu"
      orgId: 3
      dashboard:
        name: "ondemand-clusters"
        uid: "aaba6Ahbauquag"
        panels:
          cpu: 20
          memory: 24
      labels:
        cluster: "cluster"
        host: "host"
        jobid: "jobid"
```

I can work on Grafana docs in separate pull request if that's something we want to call out.

johrstrom commented:
Yea, I'd +1 that.

treydock commented:
Add a few changes. Removed beta for SELinux, mentioned Keycloak 2FA with Duo and added Grafana docs.

johrstrom commented:
Added some things and rebased off of the current 1.7 release branch because I needed to add some things close to where we added things recently and wanted to avoid conflicts.

ericfranz commented on source/customization.rst:
```diff
@@ -174,30 +178,34 @@ Start by creating the file
-- The variable ``paths`` is an array of ``Pathname`` objects that define a list
+    # Project scratch has an optional title field.
+    paths.concat projects.map { |p| FavoritePath.new(Pathname.new("/fs/scratch/#{p}"), title: "Scratch")  }
+  end
```
The indenting got modified by accident here. The code blocks should be indented with the comment, lining up with the "c" of  the "code-block" word but they were shifted back 1 character

johrstrom replied:
Should it be though?  It seems more natural to just keep the 2 space (tabs) instead of offsetting the initial indent by 3, then using indents of 2 within the block.

ericfranz replied:
Yes. The outer indentation is following Sphinx/restructured conventions. The 2 spaces in the block is following Ruby conventions.

ericfranz commented on source/customization.rst:
```diff
@@ -244,6 +252,27 @@ In ``/etc/ood/config/apps/shell/env`` set the env var ``DEFAULT_SSHHOST`` to cha
 
 This will control what host the shell app ssh's to when the URL accessed is ``/pun/sys/shell/ssh/default`` which is the URL other apps will use (unless there is context to specify the cluster to ssh to).
 
+SSH Wrapper
```
Rename "Shell App SSH Command Wrapper" and in the first sentence clarify this is for the shell app.

ericfranz commented on source/release-notes/v1.7-release-notes.rst:
```diff
@@ -79,9 +105,135 @@ Passenger has been upgraded from ``5.3.7`` to ``6.0.4``. NGINX has been upgraded
+Systemd will now regenerate the ood-portal.conf before every apache restart.  This means
+and administrators can now edit the ``ood_portal.yml`` configuration and then restart httpd
+with systemctl and apache will read the new configuration, because a new ood-portal.conf
+will have been written just before restarting.
```
Clarify that

1. this is made possible due to the use of the checksum feature and that this will not work if the checksum indicates that there have been manual changes made to the generated ood-portal.conf
2. clarify that this means when editing `ood_portal.yml` you do not need to follow up with executing ood-portal-generator script
3. clarify the exception is when installing 1.7, where you still have to execute the script with the force flag

treydock replied:
The checksum feature was introduced in 1.6, it's not new for 1.7.  So I feel like #1 is old news.  Still want #1 called out?

ericfranz replied:
Yes, I think my point is that we are leveraging that feature to support safely updating the ood-portal.conf file automatically on every Apache restart.

ericfranz commented on source/release-notes/v1.7-release-notes.rst:
```diff
@@ -79,9 +105,135 @@ Passenger has been upgraded from ``5.3.7`` to ``6.0.4``. NGINX has been upgraded
+Administrators can now set the ``OOD_JOB_NAME_ILLEGAL_CHARS`` environment variable to prevent
+characters from being used in job names.  For example if you do not want to use ``/`` in job
+names (as is the case with some :ref:`Grid Engine <resource-manager-sge>` versions) you would
+set ``OOD_JOB_NAME_ILLEGAL_CHARS: '/'`` in the ``pun_custom_env`` attribute of ``nginx_stage.yml``.
```
Maybe we should have a section in "Customization" for "Sanitizing job names" and then reference that in the Grid Engine example? This is a good example of several cases where in the release notes (1.5, 1.6, and now this) we sometimes have better feature documentation than in the main documentation area.

ericfranz commented on source/release-notes/v1.7-release-notes.rst:
```diff
@@ -79,9 +105,135 @@ Passenger has been upgraded from ``5.3.7`` to ``6.0.4``. NGINX has been upgraded
+
+The ``nginx_stage nginx_clean`` command now supports a ``-u`` or ``--user`` option so it may
+clean a specific users' PUN. For example ``nginx_stage nginx_clean -u johndoe`` would only
+clean johndoe's PUN and disregard all the others.
```
Maybe say "so it may kill a specific user's PUN"
i.e.
kill instead of clean (even though the command is called nginx_clean which is a funny name for the kill command)

ericfranz commented on source/release-notes/v1.7-release-notes.rst:
```diff
@@ -79,9 +105,135 @@ Passenger has been upgraded from ``5.3.7`` to ``6.0.4``. NGINX has been upgraded
+* Has a resource manager's directive (#PBS, #SBATCH, #BSUB or #$) in the first 1000 characters.
+
+Other valid files only have to meet a size requirement of less than ``OOD_MAX_SCRIPT_SIZE_KB`` which
+defaults to 65 (meaning 65 kb).
```
This should also have a permenant place to live in the actual documentation.

ericfranz commented on source/release-notes/v1.7-release-notes.rst:
```diff
@@ -79,9 +105,135 @@ Passenger has been upgraded from ``5.3.7`` to ``6.0.4``. NGINX has been upgraded
+node that the job is running on. This is helpful for sites that don't allow regular users to shell into 
+compute nodes.
+
+To do so, simply set ``OOD_BC_SSH_TO_COMPUTE_NODE=0`` (or 'false' or 'off') in ``/etc/ood/config/apps/dashboard/env``.
```
This should also have a permanent place to live in the actual documentation.
